runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -547,6 +547,26 @@ public static UUID getStreamIdForStreamTag(String namespace, String streamTag) {
         return (Table<K, V, M>) tableMap.get(fullyQualifiedTableName);
     }
 
+    /**
+     * Close a table that is already opened.
+     * Throws a NoSuchElementException if table is not previously opened and not present in cache.
+     *
+     * @param namespace Namespace of the table.
+     * @param tableName Name of the table.
+     * @return true if a table was found and removed from cache, false otherwise
+     */
+    public boolean closeTable(String namespace, String tableName) {
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+        Table<Message, Message, Message> table = tableMap.get(fullyQualifiedTableName);
+        if (table == null) {
+            log.warn("closeTable: Did not find any table {}", fullyQualifiedTableName);
+            return false;
```

Comment:
any reason why we need a return value for this function?  If the table was not opened, it will be a no-op.  Is it expected that clients will use the return value for something?

Reply:
Yes, in case they come back and ask us for an api which tells us if a table was removed from cache successfully. This return value just avoids us having to do another change just for that ask.

Reply:
mmm... do you think it is likely that there will be such as ask?  My suggestion is we throw an exception whenever the table cannot be closed(not found or another runtime error) instead of a return value.

Reply:
In my opinion I don't think we should throw an exception. If you think of it, the 'closeTable' aims to be an optimization, so if the table is not even opened it should just be a no-op, i.e., there is nothing to clear for optimized performance.